src/Microsoft.AspNetCore.SignalR/DynamicHub.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.SignalR
+{
+    public class DynamicHub : Hub
+    {
+        private dynamic _clients;
+
+        public new DynamicHubClients Clients
+        {
+            get { return _clients ?? new DynamicHubClients(base.Clients); }
```

Comment:
Add caching for the property so that DynamicHubClients isn't created every time? Though tbh, this only exists for the hub invocation.